cmake/templates/BoostConfig.cmake.in

```diff
@@ -102,3 +102,12 @@ if(TARGET Boost::iostreams)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostZLIBDependency.cmake" OPTIONAL)
   include("${CMAKE_CURRENT_LIST_DIR}/BoostBZip2Dependency.cmake" OPTIONAL)
 endif()
+
+if(TARGET Boost::thread)
+  find_package(Threads)
+  set_target_properties(
+    Boost::thread
+    PROPERTIES
+    INTERFACE_LINK_LIBRARIES "${CMAKE_THREAD_LIBS_INIT}"
```

Comment:
We already have `set_target_properties(... PROPERTIES INTERFACE_LINK_LIBRARIES Boost::boost")` so I think we should append property (read, modify, set).